### ISSUE TITLE
Support "dumping" models into their correct json representation

### ISSUE THREAD

**Issue by patkujawa-wf** [rbanffy/appengine-fixture-loader#7]
It seems to me that the next logical step would be to allow one to take an existing model, in code, and serialize it to a json file suitable for consumption by this library. Or at least add a helper method to get 80% of the way there :)

**Comment by rbanffy** [comment]
I agree, but there would be some complications. Server processes cannot write to the server's filesystem (which is ephemeral, anyway), so we would be restricted to dumping local data. Local data would be from a running server, but that server data is relatively ephemeral too - the default setting is to keep the data under /tmp and, therefore, it doesn't survive a reboot. I can see some uses for dumping the current working dataset of an application during debugging - I've used the Django admin to generate data that would later be used for testing - but I don't see this as a common usage scenario on App Engine development.

**Comment by patkujawa-wf** [comment]
Those are good points. I was thinking it might just reduce the barrier to
entry if one could start up there local server and have some code grab
existing entities and dump them as json. For instance, a temporary endpoint
that just returned the response as json. Then the user manually saves that
file to their file system.

On Tue, Feb 10, 2015 at 4:52 PM, Ricardo Bánffy <notifications@github.com>
wrote:

> I agree, but there would be some complications. Server processes cannot
> write to the server's filesystem (which is ephemeral, anyway), so we would
> be restricted to dumping local data. Local data would be from a running
> server, but that server data is relatively ephemeral too - the default
> setting is to keep the data under /tmp and, therefore, it doesn't survive a
> reboot. I can see some uses for dumping the current working dataset of an
> application during debugging - I've used the Django admin to generate data
> that would later be used for testing - but I don't see this as a common
> usage scenario on App Engine development.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rbanffy/appengine-fixture-loader/issues/7#issuecomment-73809863>
> .
>



-- 
*Pat Kujawa*

*Software Engineer*

*Workiva Inc.*
*Missoula, MT*

**Comment by rbanffy** [comment]
It could take the form of a command-line utility using the same plumbing `download_data` uses, but it would be a little complicated to replicate the `__children__[x]__` thing. It could, however, use the explicit `__key__` declaration to express the same hierarchies in a different way. The first step, however, would be to implement `__key__

**Comment by patkujawa-wf** [comment]
And even a rough MVP that didn't handle hierarchies would be useful. I've seen models with hundreds of properties (e.g. for global settings), and no one wants to hand-write those :)

**Comment by rbanffy** [comment]
Makes sense. We should think about it.
